api/pkg/db/migration/202107291608_refresh_all_tables.go

```diff
@@ -0,0 +1,262 @@
+// Copyright © 2021 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package migration
+
+import (
+	"github.com/go-gormigrate/gormigrate/v2"
+	"github.com/tektoncd/hub/api/gen/log"
+	"github.com/tektoncd/hub/api/pkg/db/model"
+	"gorm.io/gorm"
+)
+
+func refreshAllTables(log *log.Logger) *gormigrate.Migration {
+	return &gormigrate.Migration{
+		ID: "202107291608_refresh_all_tables",
+		Migrate: func(db *gorm.DB) error {
+			var user []model.User
+			if err := db.Find(&user).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var user_scopes []model.UserScope
+			if err := db.Find(&user_scopes).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var user_rating []model.UserResourceRating
+			if err := db.Find(&user_rating).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var tags []model.Tag
+			if err := db.Find(&tags).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var sync_job []model.SyncJob
+			if err := db.Find(&sync_job).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var scopes []model.Scope
+			if err := db.Find(&scopes).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var resources []model.Resource
+			if err := db.Find(&resources).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var resource_versions []model.ResourceVersion
+			if err := db.Find(&resource_versions).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var resource_tags []model.ResourceTag
+			if err := db.Find(&resource_tags).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var configs []model.Config
+			if err := db.Find(&configs).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var catalog []model.Catalog
+			if err := db.Find(&catalog).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			var catalog_error []model.CatalogError
+			if err := db.Find(&catalog_error).Error; err != nil {
+				log.Error(err)
+				return err
+			}
+
+			txn := db.Begin()
+			if err := db.Migrator().DropTable(&model.Category{},
```

Comment:
we are starting a `txn` but not using them while executing queries then how it would be helpful 😸

Reply:
https://gorm.io/docs/transactions.html#Transactions-by-manual

Reply:
```
txn := db.Begin()
if err:= refreshDb(txn, log); err!=nil{
  log error
  txn.rollback()
}
txn.commit
```
and use txn passed for all queries